cmd/compute-domain-daemon/computedomain.go

```diff
@@ -197,7 +197,11 @@ func (m *ComputeDomainManager) UpdateComputeDomainNodeInfo(ctx context.Context,
 
 // BlockUntilAllNodesJoinComputeDomain waits until all nodes have joined the compute domain
 // and returns the list of nodes in the compute domain.
-func (m *ComputeDomainManager) BlockUntilAllNodesJoinComputeDomain() []*nvapi.ComputeDomainNode {
-	<-m.nodesChan
-	return m.nodes
+func (m *ComputeDomainManager) BlockUntilAllNodesJoinComputeDomain(ctx context.Context) ([]*nvapi.ComputeDomainNode, error) {
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
```

Comment:
Is [this](https://github.com/NVIDIA/k8s-dra-driver-gpu/blob/3e825113b271f935eaee7d9c5cbe192fd6fc15bc/cmd/compute-domain-daemon/main.go#L74C4-L74C12) where `ctx` is canceled in response to SIGTERM?

Reply:
yep, thats it